src/Symfony/Component/HttpKernel/Controller/LegacyArgumentResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Responsible for the creation of the action arguments.
+ *
+ * @deprecated This class is deprecated since 3.1 and will be removed in 4.0. Please use the ArgumentResolver::class instead.
```

Comment:
Yeah but using ``@deprecated`` means to me that this class can be removed until 4.0 (it can't be renamed for example) so someone could use it instead of extending the ``ControllerResolver`` or using the new system.
Anyway I have no strong opinion about it, this just looks strange to me but that's not important.